src/Microsoft.AspNetCore.ApplicationInsights.HostingStartup/ApplicationInsightsStartupLoader.cs

```diff
@@ -22,6 +24,16 @@ public class ApplicationInsightsHostingStartup : IHostingStartup
         public void Configure(IWebHostBuilder builder)
         {
             builder.UseApplicationInsights();
+            builder.ConfigureServices(InitializeServices);
+        }
+
+        /// <summary>
+        /// Adds the Javascript <see cref="TagHelperComponent"/> to the <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> associated with the application.</param>
+        protected virtual void InitializeServices(IServiceCollection services)
```

Comment:
Generally on callbacks we don't do arg validation because it is assumed that the caller is well-behaved.

Though, I believe that this particular package is never referenced by apps - I don't know that we even publish it on NuGet.org to begin with. As such, perhaps having it private is sufficient.